### PR TITLE
ansible: add 10.15 macs

### PR DESCRIPTION
**Pull request by AshCripps**
Its finally here 🎉 

Currently a draft as ive only had time to add one machine and shave a couple hairy yaks

Fixes: #1695

### CODE REVIEW

File: ansible/roles/java-base/tasks/main.yml [hunk 1]
```diff
@@ -98,7 +98,10 @@
 - name: install java (macOS)
   become_user: administrator
   when: java.rc > 0 and os|startswith("macos")
-  package: name="{{ java_package_name }}" state=present
+  homebrew_cask:
+        name: "{{ java_package_name }}"
+        state: present
+  #package: name="{{ java_package_name }}" state=present
```
Comment (sam-github) [inline ansible/roles/java-base/tasks/main.yml]: delete?

Comment (AshCripps) [reply ansible/roles/java-base/tasks/main.yml]: > LGTM, though I understand its WIP. There was more churn in inventory.yml than I expected, does it include some unrelated changes, or is it some cleanups while you are there?

Some cleanup I noticed the release section wasn't in order

File: ansible/roles/java-base/tasks/main.yml [hunk 2]
```diff
@@ -95,10 +95,20 @@
     - java.rc > 0 and not os|startswith("macos")
   package: name="{{ java_package_name }}" state=present
 
+- name: install java tap (macOS)
+  become_user: administrator
+  when: java.rc > 0 and os|startswith("macos")
+  homebrew_tap:
+        name: AdoptOpenJDK/openjdk
+        state: present
```
Comment (AshCripps) [inline ansible/roles/java-base/tasks/main.yml]: This is effectivly duplicated code from https://github.com/nodejs/build/blob/1cbb6cf82aaa2646965ff2c15128220ecfe4de16/ansible/roles/package-upgrade/tasks/partials/brew.yml#L36-L39 but I needed this change to get be able install the openjdk, even though ansible tells me the task ive linked to ran ok. @nodejs/build anyone have an ideas/experience as to why this code works in one place but not another?

**Comment by AshCripps** [comment]
Created the jenkins agents - https://ci.nodejs.org/label/osx1015/

Will need someone to create the release agent for me.

@nodejs/build-infra can someone add the IPs to the respective CI firewalls please?

**Comment by AshCripps** [comment]
Currently seeing these test failures on one of the test machines:

```
=== release test-dgram-connect-send-empty-array ===
Path: parallel/test-dgram-connect-send-empty-array
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-connect-send-empty-array.js
--- TIMEOUT ---
=== release test-dgram-connect-send-empty-buffer ===
Path: parallel/test-dgram-connect-send-empty-buffer
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-connect-send-empty-buffer.js
--- TIMEOUT ---
=== release test-dgram-connect-send-empty-packet ===
Path: parallel/test-dgram-connect-send-empty-packet
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-connect-send-empty-packet.js
--- TIMEOUT ---
=== release test-dgram-send-empty-array ===
Path: parallel/test-dgram-send-empty-array
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-send-empty-array.js
--- TIMEOUT ---
=== release test-dgram-send-empty-buffer ===
Path: parallel/test-dgram-send-empty-buffer
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-send-empty-buffer.js
--- TIMEOUT ---
=== release test-dgram-send-empty-packet ===
Path: parallel/test-dgram-send-empty-packet
Command: out/Release/node /Users/iojs/node/test/parallel/test-dgram-send-empty-packet.js
--- TIMEOUT ---
[11:32|% 100|+ 2913|-   6]: Done
```

Im assuming these are errors due to some mising network settings?

**Comment by sam-github** [comment]
Perhaps a one-time run of https://github.com/nodejs/node/blob/master/tools/macos-firewall.sh is needed?

**Comment by AshCripps** [comment]
Done some digging, looks like this is a known catalina issue - https://github.com/nodejs/node/issues/30030 which Apple has not fixed yet

**Comment by sam-github** [comment]
@AshCripps can you try https://github.com/nodejs/node/pull/31936 and see if the updating of expected status fixes this for you?

File: ansible/roles/java-base/tasks/main.yml (hunk 2 above)
Comment (rvagg) [reply ansible/roles/java-base/tasks/main.yml]: So partials/brew.yml ran and you saw it running homebrew_tap for the AdoptOpenJDK/openjdk cask and the next task, which installs the package, failed until you yanked the homebrew_tap into this position? It's not an ordering problem is it, partials/brew.yml ran before this didn't it?

Comment (AshCripps) [reply ansible/roles/java-base/tasks/main.yml]: Yeah I can see partials/brew,yml run but it reports nothing changed under `add AdoptOpenJDK repo` it wasnt until I add the tap to this position that anything happened

**Comment by AshCripps** [comment]
> sgtm, do I take it from this that release-nearform-macos10.15-x64-1 is a thing and we could start switching releases over to it soon?

yes @rvagg there is a release machine setup, just waiting on its IP to be added to the firewall - same as the test machines.

need https://github.com/nodejs/node/pull/31936 to land on all lines to stop the constant failures

**Comment by mhdawson** [comment]
@AshCripps firewall entries added for both test and release.

**Comment by sam-github** [comment]
@AshCripps the machine is in ci-release and is connected: https://ci-release.nodejs.org/computer/release-nearform-macos10.15-x64-1/ (for those with visibility).

I had trouble getting in by ssh, but could you add the release team ssh key to .ssh/authorized_keys?

```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDFCAmx+ldk/UyR0x0xrsvdORi59uGPdErcitIjjBZRFnE87MGIZtmsZIqGogxgEc1A1t86/eCCvL7we31iE0sF71Stx17ZxTKuo/VCTACeg1GRd3/wdVHXphAss5sBYyV/T7hcGq7aq4PsrloAUhLfmivXAG0ALhWG3AAv4AnVWaa372XWIoYqiDQCL5uEClEP/A95CdqRrbYY4bWZRR0eZMgWhX5Npy1tThuA+p0+mmvY6AGHpTW33+d8qfGwubv1tPvV7G+vQ2cjlQTWd+NfnmzoOGUIcXGZ8yhfdXwFeNdduyIalL1Jm2WxbyStQ2SFXIFgSK+tjVEMGC4H+N8oOd2jbZYANxkCDugbaP3OSYwOYyUGnMJztt1g8oemGN9QtY7GGssuOxeUrYXGAYpD7BEqX7Jh2v5zd7Uvl7W60oYwikTtJkIEclEKOE2dEhHwDrZbt3KhF+VSjhaZdfrt+fMm3qQMGPdx16lT/IxHUm825A6Nzbmtm0H8yczXGnhG59w3bGAcUxD7OqZPmQaQnw1Abba0BBbsjj28KdkUddm91DjRzJCnXtJ6DLJxxn40MISE/loy/yvCLb3HCvhgo55hB4C0XCeMfbVYGwbx3TrX1en89aZ96s9EAMATZk4S+47cnx3jfy/5CP6ms4UsvGCelMPhgdqGBG/xOr7kKw== release@build.nodejs.org
```

^--- check for inadvertent cut-n-paste line breaks! should be one line. That's the public key, it has no sec significance, but once its there any release team member can ssh in (I assume, not a member myself).

At that point, I think @rvagg could dry-run notarization.

**Comment by AshCripps** [comment]
@sam-github added, can you try now?

**Comment by sam-github** [comment]
@AshCripps no, I can't try, only [these people](https://github.com/nodejs-private/secrets/tree/master/build/release/.gpg) have the private key that goes with that public one :-)

I'll open an issue to get you enough access to do add  the machines to test CI.

**Comment by sam-github** [comment]
@AshCripps you are already a https://github.com/nodejs/build#jenkins-admins, so, the thing to do now is to create a node-test-commit-osx-ashcripps job by cloning the base job. Once you have the clone, go to the machine selection part, and find the osx1015 label that the 3 new test machines have, and click it. At that point, you should be able trigger a run of the job against nodejs/master from the Jenkins UI, and see that they fail on the tests that are skipped by https://github.com/nodejs/node/pull/31936, or you could even test my specific branch from that PR, if you want, to confirm the build is green. Then, once we can see the build is green in your cloned job, we are good to select the osx1015 machines to add them to CI in the regular node-test-commit-osx... and the test side of this is done.

After that, its mop-up: adding the label to the release builds, the v8 builds, citgm.... etc.

**Comment by AshCripps** [comment]
build of master: https://ci.nodejs.org/job/node-test-commit-osx-AshCripps/1/nodes=osx1015/

build of @sam-github's PR: https://ci.nodejs.org/job/node-test-commit-osx-AshCripps/1/nodes=osx1015/

**Comment by AshCripps** [comment]
oops @rvagg the release machine needs this doing to it - https://github.com/nodejs/build/issues/1393. I dont have the access nessacary to do it I dont think

**Comment by AshCripps** [comment]
Added the machines to `node-test-commit` and all `citgm` jobs

Tried adding it to a libuv job but I get an xcodebuild error which is weird as xcodebuild isnt on any of the currently working machines: https://ci.nodejs.org/job/libuv-test-commit-osx-AshCripps/1/

Any ideas @richardlau?

**Comment by richardlau** [comment]
It’s probably related to https://github.com/nodejs/build/issues/2172 and the version of gyp being used being old. gyp in nodejs/node and nodejs/node-gyp were updated to cope with double digit versions of xcode.

**Pull request merged**